crates/shared/src/price_estimation/competition.rs

```diff
@@ -203,17 +215,16 @@ impl PriceEstimating for RacingCompetitionPriceEstimator {
 
                 if let Some(competition) = &self.competition {
                     let trade = Trade::from(&*query);
-                    let estimator = EstimatorIndex(*estimator_index);
+                    let estimator = EstimatorIndex(*stage_index, *estimator_index);
                     let was_correct = predictions.iter().any(|p| p.winner == estimator);
                     metrics().record_prediction(&trade, was_correct);
                     competition.record_winner(trade, estimator);
                     metrics()
                         .requests
-                        .with_label_values(&["saved"])
+                        .with_label_values(&["executed"])
```

Comment:
The idea of the metrics was to count how many requests we would not have to execute because of the winner prediction logic (hence increment by `all_possible_requests - number_of_predictions`). If we want to keep that logic mostly untouched I would suggest to keep the metric label and set `requests` to the total number of all estimators (not the number of stages).

But I think over all the current prediction model doesn't fit very well with the staggered approach.
E.g. let's say the prediction is that most of the time we need a stage 3 estimator and sometimes a stage 2 estimator. If we then want to skip requests based on these requests the ordering of the stages has to be bypassed entirely for the "optimised" request issue pattern. And if that fails we'd have to do the vanilla logic while avoiding to send duplicated requests for estimators we already sent requests to based on the winner prediction.
I believe if we want to explore the prediction approach further in the future we have to rethink it so we could already drop the current implementation with this PR to reduce unused complexity.